### ISSUE TITLE
correctif fill_blanks.class.php

### ISSUE THREAD

**Issue by alexiye** [chamilo/chamilo-lms#2415]
### Current behavior / Resultado actual / Résultat actuel
![fillblanksavantcorrection](https://user-images.githubusercontent.com/24505996/37154142-5dd62be6-22df-11e8-8947-8223cbf12a77.png)
-corriger ligne 1279  
feedback-red devient feedback-green
-corriger ligne 1289  
il manque un point virgule après feedback-question
Après la correction :
![fillblanksaprescorrection](https://user-images.githubusercontent.com/24505996/37154296-cfb2e24a-22df-11e8-8e31-7d4ee54c96ff.png)

**Comment by jmontoyaa** [comment]
This part of the code has been change, it works well for me. Can you try again?
Better if you can make an example of the error here: 

https://11.chamilo.org/ 

login/pass: admin/admin

**Comment by alexiye** [comment]
https://11.chamilo.org/main/exercise/exercise_result.php?cidReq=FILLBLANKS&id_session=0&gidReq=0&gradebook=0&origin=&exe_id=613&learnpath_id=0&learnpath_item_id=0&learnpath_item_view_id=0
![resultats](https://user-images.githubusercontent.com/24505996/37339964-b11c6f50-26bc-11e8-88f8-e8cbb394f3e4.png)
Quand on fait prévisualiser l'exercice, on ne voit pas le menu déroulant [x|y|z]
![on ne voit pas le blanc en menu deroulant](https://user-images.githubusercontent.com/24505996/37339996-d1b3e234-26bc-11e8-8b14-c24636798d8d.png)

**Comment by jmontoyaa** [comment]
I sent a fix for the preview, your exercise should contain only one "|" not double "||"

henry||HENRY||Henry

**Comment by alexiye** [comment]
double || c'est quand il y a plusieurs reponses possibles non ?

Alexandre Bigan

**Comment by jmontoyaa** [comment]
Oui, tu as raison. j'ai parlé avec @baelmyhu l'expert de fill in blanks et il m'a raconté comment ca marche.

Je peux réproduire l'erreur maintenant.

**Comment by jmontoyaa** [comment]
J'ai corrigé l'erreur.

**Comment by alexiye** [comment]
1290 $result = "";
ne manque-t-il pas un point virgule pour éviter le saut à la ligne dans la correction :
 
1290 $result = "";
 
> Message du 14/03/18 15:11
> De : "Julio Montoya" 
> A : "chamilo/chamilo-lms" 
> Copie à : "alexiye" , "Author" 
> Objet : Re: [chamilo/chamilo-lms] correctif fill_blanks.class.php (#2415)
> 
>
> J'ai corrigé l'erreur.
—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or mute the thread.



{"api_version":"1.0","publisher":{"api_key":"05dde50f1d1a384dd78767c55493e4bb","name":"GitHub"},"entity":{"external_key":"github/chamilo/chamilo-lms","title":"chamilo/chamilo-lms","subtitle":"GitHub repository","main_image_url":"https://cloud.githubusercontent.com/assets/143418/17495839/a5054eac-5d88-11e6-95fc-7290892c7bb5.png","avatar_image_url":"https://cloud.githubusercontent.com/assets/143418/15842166/7c72db34-2c0b-11e6-9aed-b52498112777.png","action":{"name":"Open in GitHub","url":"https://github.com/chamilo/chamilo-lms"}},"updates":{"snippets":[{"icon":"PERSON","message":"@jmontoyaa in #2415: J'ai corrigé l'erreur. "}],"action":{"name":"View Issue","url":"https://github.com/chamilo/chamilo-lms/issues/2415#issuecomment-373034352"}}}

**Comment by jmontoyaa** [comment]
Ce changement n'est pas un CSS valide ... 

C'est à cause de "display: block;" 

<pre>
.feedback-question {
    margin-top: 5px;
    margin-bottom: 5px;
    // display: block;
}</pre>

**Comment by AngelFQC** [comment]
This style for question was created for ofaj branch and the images should not be here but was added to 1.11.x here https://github.com/chamilo/chamilo-lms/commit/531f5aa24e3f66c730bf48d2cd78f8e950c3364c#diff-8b186ddd6e577c60ef19e31036be3b37R7330
I sent a minor style change

**Comment by AngelFQC** [comment]
@aragonc sería mejor quitar las imágenes que parecen radiobuttons o mejorar el color para que vaya con el tema de chamilo. Aunque inicialmente estas imágenes no estaban en 1.11.x

**Comment by jmontoyaa** [comment]
Si se hacen cambios confirmarlos con @NicoDucou para que se integren bien en ofaj_dev
La idea es que todo lo que va a 1.11.x será pasado a ofaj_dev.

**Comment by alexiye** [comment]
Je suis désolé mais j'ai de nouvelles erreurs que je n'arrive pas à comprendre.
Celles-ci sont visible dans le lien suivant :

https://11.chamilo.org/main/exercise/exercise_result.php?cidReq=LEPASSECOMPOSECE2&id_session=0&gidReq=0&gradebook=0&origin=&exe_id=626&learnpath_id=0&learnpath_item_id=0&learnpath_item_view_id=0 

Quand plusieurs réponses sont possibles j'utilise ||
Cela fonctionne pour je suis [venu||venue] mais des erreurs surviennent avec tu es [allé||allée]

**Comment by alexiye** [comment]
J'ai vu que cela a été corrigé, merci !

**Comment by aragonc** [comment]
@jmontoyaa creo que ahora es un icono fijo, se tendría que ver como hacer el cambio para no afectar a ofaj_dev, pensé que no se integraría a Chamilo? @NicoDucou el ofaj_dev esta actualizado con la 1.11.x actual?

**Comment by jmontoyaa** [comment]
Lo mejor sería hacerlo por css, cosa que el cambio afecta a base.css pero es modificable en el estilo de ofaj.

El portal de ofaj está parcialmente actualizado a 1.11.x.

**Comment by aragonc** [comment]
Lo envié
![check](https://user-images.githubusercontent.com/2940509/38838401-89e1a898-419b-11e8-8576-de31376a5a8c.png)
 a la rama 1.11.x quedaría asi, en ofaj faltaría revisarlo, aunque creo que esos iconos están en la carpeta del theme que ofaj usa.
Por cierto me olvide colocar el número de issue en el commit

**Comment by jmontoyaa** [comment]
El "check" debería estar en verde y la "cruz" rojo.

**Comment by aragonc** [comment]
He mejorado la vista, separando la respuesta correcta del sistema con la del usuario, adicionalmente las imágenes de los check y error verde o rojo según corresponda.

![feedback](https://user-images.githubusercontent.com/2940509/38939167-ebfbdb50-42ec-11e8-9afb-8c91a59ea5b1.png)

**Comment by alexiye** [comment]
Le style ne s'applique pas tout le temps ! Sinon c'est super.
https://11.chamilo.org/main/exercise/exercise_show.php?cidReq=LACCORDENGENREDELADJECTIFQUALIFIC&id_session=0&gidReq=0&gradebook=0&origin=&action=qualify&id=653

**Comment by jmontoyaa** [comment]
Buena mejora, pero en el screenshot que muestras se destaca más lo gris de todo el texto. Tal vez bajarle un poco el gris y hacer llamativo el rojo y verde asi:

![download](https://user-images.githubusercontent.com/158935/38974972-5653651c-43ac-11e8-9764-fd127872b453.png)

**Comment by aragonc** [comment]
Cambio de color aplicado... también a listas despegables.
![color](https://user-images.githubusercontent.com/2940509/39134214-b24757d4-46db-11e8-9ef8-7ea087636290.png)

**Comment by alexiye** [comment]
Bravo !

**Comment by jmontoyaa** [comment]
Me parece que ese verde es poco legible. Tratar de ponerle un fondo verde y texto claro o obscuro.
Por ejemplo, el texto en azul claro en tu screenshot esta muy bien!

**Comment by alexiye** [comment]
Le style ne s'applique pas quand on peut avoir plusieurs solutions possibles en utilisant "||"

**Comment by alexiye** [comment]
:+1:
